### PR TITLE
Adds test to AuctionViewController

### PR DESCRIPTION
**Pull request by ashfurrow**
This adds tests for the `AuctionViewController`. How? Let's break it down.

The auctions VC was referring directly to the `AuctionNetworkModel`, which is bad because we want to stub it (_without_ subclassing, ew). So we confine the VC's reference to it to be a protocol that exposes just the functionality the VC needs. Then in our test, we create a new class that conforms to that protocol, but returns predictable results. Then we just DI that class in.

There's some repetition with the `RefineViewController`'s tests, and I think it's worth looking into a Quick/Nimble-equivalent of `itHasSnapshotsForDevicesWithName` that'll stub the device and size class and everything. I'll open an issue.

### CODE REVIEW

File: Artsy_Tests/Supporting_Files/ARTestContext.m [hunk 1]
```diff
@@ -12,7 +12,7 @@ - (CGRect)_applicationFrameForInterfaceOrientation:(long long)arg1 usingStatusba
 
 @implementation ARTestContext
 
-+ (void)useDevice:(enum ARDeviceType)device :(void (^)(void))block
++ (void)useDevice:(enum ARDeviceType)device:(void (^)(void))block
```
Comment (orta) [inline Artsy_Tests/Supporting_Files/ARTestContext.m]: sneaky bit of work here for space commander

Comment (ashfurrow) [inline Artsy_Tests/Supporting_Files/ARTestContext.m]: ha, yeah didn't notice that. Want me to revert?

Comment (orta) [inline Artsy_Tests/Supporting_Files/ARTestContext.m]: aye, you'll have to do the no-verify thing or whatever

**Comment by ArtsyMobile** [comment]
&nbsp; | 1 Error
    ------------- | ------------
        :no_entry_sign: | No CHANGELOG changes made
    

  

  
    &nbsp; | 1 Warning
    ------------- | ------------
        :warning: | Needs testing on a Phone if change is non-trivial
    

  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

**Comment by orta** [comment]
Could tag this a `#trivial` in the pr title if you want danger to ignore the changelog requirement?

**Comment by ashfurrow** [comment]
:green_apple:

**Comment by alloy** [comment]
Thanks!

**Comment by ashfurrow** [comment]
My pleasure! It's fun to get into VC testing and Swift, especially in an app that's much more complex than Eidolon. Lots of things to learn and turn into patterns :smile_cat:

**Pull request merged**